### PR TITLE
Features dev

### PR DESCRIPTION
**Pull request by aioaneia**

### CODE REVIEW

File: docs/source/benchmarking/benchmark.rst
Comment (HamedBabaei) [inline docs/source/benchmarking/benchmark.rst:1]: Can we also change the title of this section in the docs also from `Agricultural` to `Agriculture`?

File: ontolearner/base/learner.py [hunk 1]
```diff
@@ -17,27 +16,29 @@ def train(self, train_data: Any, task: str):
...
+class AutoLLM(ABC):
     def __init__(self):
         self.model = None
         self.tokenizer = None
 
-    def load(self, model_id:str):
+    def load(self, model_id: str):
         pass
 
     def generate(self, inputs: List[Any]):
```
Comment (HamedBabaei) [inline ontolearner/base/learner.py:33]: Can we specify the output that this `generate` function will return? is it a `List` or `Dict` or what?

File: ontolearner/base/learner.py [hunk 2]
```diff
@@ -46,6 +47,7 @@ def index(self, inputs: List[Any]):
     def retrieve(self, inputs: List[Any], top_k: int):
```
Comment (HamedBabaei) [inline ontolearner/base/learner.py:47]: Can we specify the output that this `retrieve` function will return? is it a List or Dict or what?

File: ontolearner/evaluation/metrics.py [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+    elif task == "non-taxonomy-discovery":
+        metrics['avg_exact_match'] = np.mean([r['exact_match'] for r in results])
+        metrics['avg_similarity'] = np.mean([r['similarity_score'] for r in results])
+
+    metrics['num_samples'] = len(results)
+
+    return metrics
+
+
+def plot_model_comparison(summary_df, path: Path):
```
Comment (HamedBabaei) [inline ontolearner/evaluation/metrics.py:105]: Visualization should not appear here (inside the metrics), please make another script in the `evaluation` dir and give it an appropriate name!

File: ontolearner/evaluation/metrics.py [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+
+def calculate_non_taxonomy_metrics(predicted: str, ground_truth: str) -> Dict[str, float]:
+    """
+    Calculate metrics for non-taxonomy relation prediction
+    """
+    # Exact match between relation types
+    exact_match = 1.0 if predicted.lower() == ground_truth.lower() else 0.0
+
+    # String similarity (optional if you want to give partial credit)
+    from difflib import SequenceMatcher
```
Comment (HamedBabaei) [inline ontolearner/evaluation/metrics.py:68]: Please make the import at the top of the script and not here!

File: ontolearner/evaluation/metrics.py
Comment (HamedBabaei) [inline ontolearner/evaluation/metrics.py:1]: Here are a couple of recommendations for this metrics.py

1. This is one of the evaluation script that hold the metrics, so why not call the functions as a `precision_score`, `recall_score`, `f1_score`, `exact_match` and ... the task can be given in the arguments so the calculation happens there. So, this script should only hold the metrics calculation and nothing else.
2. for aggregated metrics, you can build another script called `evaluate.py` and put this function there with `evaluation_report()` 
3. Visualization can not be defined here, it is not a metric that has been used in the evaluation it just a helper so please move it into the `evaluate.py` script!

Comment (HamedBabaei) [inline ontolearner/evaluation/metrics.py:1]: The evaluation dir misses `__init__.py`!

File: ontolearner/learner/llm.py [hunk 5]
```diff
@@ -1,12 +1,29 @@
...
 
-class AutoLLM(AutoLearnerLLM):
+
+class AutoLearnerLLM(AutoLLM):
     def __init__(self):
         super().__init__()
 
-    def load(self, model_id:str):
-        pass
+    def load(self, model_id: str = "mistralai/Mistral-7B-Instruct-v0.1"):
```
Comment (HamedBabaei) [inline ontolearner/learner/llm.py:11]: What about the token? If I use the llama model that asks me a Huggingface token, what will happen here? so I would recommend considering a token as a kind of variable that is set to `token:str=' '"` in the init function of this model. And just pass it to the tokenizer and automodelforcausallm they will operate if token is just a empty string!

File: ontolearner/utils/train_test_split.py [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+from typing import Tuple, List, Set
+import logging
+from sklearn.model_selection import train_test_split as sklearn_split
+
+from ontolearner.data_structure import OntologyData, TermTyping, TaxonomicRelation, NonTaxonomicRelation
+
+logger = logging.getLogger(__name__)
+
+
+def stratified_term_typing_split(term_typings: List[TermTyping], test_size: float = 0.2, random_state: int = 42) \
```
Comment (HamedBabaei) [inline ontolearner/utils/train_test_split.py:11]: @aioaneia please rename this as a `term_typing_split`

File: ontolearner/utils/train_test_split.py [hunk 7]
```diff
@@ -0,0 +1,219 @@
...
+        n_test = max(1, int(len(train_relations) * test_size))
+        random.Random(random_state).shuffle(train_relations)
+        test_relations = train_relations[-n_test:]
+        train_relations = train_relations[:-n_test]
+
+    logger.info(f"Taxonomy split: {len(train_relations)} train, {len(test_relations)} test")
+    return train_relations, test_relations
+
+
+def non_taxonomy_split(
```
Comment (HamedBabaei) [inline ontolearner/utils/train_test_split.py:104]: rename this function as a `non_taxonomic_re_split`

File: ontolearner/utils/train_test_split.py [hunk 8]
```diff
@@ -0,0 +1,219 @@
...
+            test_candidate_relations.remove(rel)
+
+    # Add remaining candidates to train
+    train_relations.extend(test_candidate_relations)
+
+    logger.info(f"Non-taxonomy split: {len(train_relations)} train, {len(test_relations)} test")
+    return train_relations, test_relations
+
+
+def ontology_train_test_split(data: OntologyData, test_size: float = 0.2, random_state: int = 42) \
```
Comment (HamedBabaei) [inline ontolearner/utils/train_test_split.py:150]: just `train_test_split`

**Review by HamedBabaei — COMMENTED** [review]
@aioaneia please can you adapt these minor changes? it is a bit urgent
